lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -786,15 +801,53 @@ listPools spl =
             apparentPerformance
 
 joinStakePool
-    :: forall n k.
+    :: forall ctx s t n k.
         ( PaymentAddress n k
+        , Buildable (ErrValidateSelection t)
+        , s ~ SeqState n k
         , k ~ ShelleyKey
+        , HardDerivation k
+        , ctx ~ ApiLayer s t k
         )
-    => ApiT PoolId
+    => ctx
+    -> StakePoolLayer IO
+    -> ApiT PoolId
     -> ApiT WalletId
     -> ApiWalletPassphrase
     -> Handler (ApiTransaction n)
-joinStakePool _ _ _ = throwError err501
+joinStakePool ctx spl (ApiT poolId) (ApiT wid) passwd = do
+    (_, walMeta, _) <- liftHandler $ withWorkerCtx ctx wid throwE $
+        \wrk -> W.readWallet wrk wid
+    when (walMeta ^. #delegation == W.Delegating poolId) $
+        liftHandler $ throwE (ErrCreateCertPoolAlreadyJoined poolId)
+
+    allPools <- listPools spl
+    case L.find (\(ApiStakePool pId _ _) -> pId == ApiT poolId) allPools of
+        Nothing ->
+            liftHandler $ throwE (ErrCreateCertNoSuchPool poolId)
+        Just _ -> do
+            let (ApiWalletPassphrase (ApiT pwd)) = passwd
+
+            selection <- liftHandler $ withWorkerCtx ctx wid liftE1 $ \wrk ->
+                W.createCert @_ @s @t wrk wid
+
+            _ <- throwError err501
```

Comment:
**[suggestion: potentially error prone]**
Remove this line.

**Why**
`signCert` will already throw loudly and cause the request to fail with a 500. So this looks redundant to me. Otherwise, we need to remember to modify this section of the code after implementing `signCert`. This could be confusing, although, this is very minor.

Reply:
indeed, done